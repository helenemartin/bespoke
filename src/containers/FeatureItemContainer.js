import { useSpeechSynthesis } from "react-speech-kit";
import Feature from "../components/feature";

export function FeatureItemContainer({ item }) {
  const { speak } = useSpeechSynthesis();

  return (
    <>
      <Feature key={item.id} direction={item.direction}>
        <Feature.Pane>
          <Feature.Title data-testid="title">{item.title}</Feature.Title>
          <Feature.SubTitle>{item.subTitle}</Feature.SubTitle>
          <Feature.Button
            data-testid="clickable-feature"
            onClick={() => speak({ text: item.spokenText })}
          >
            ðŸ‘„
          </Feature.Button>
        </Feature.Pane>
        <Feature.Pane>
          <Feature.Flipper>
            <Feature.InnerFrame>
              <Feature.Front>
                <Feature.Image src={item.image} alt={item.alt} />
              </Feature.Front>
              <Feature.Back>
                <Feature.Image src={item.image2} alt={item.alt} />
              </Feature.Back>
            </Feature.InnerFrame>
          </Feature.Flipper>
        </Feature.Pane>
      </Feature>
    </>
  );
}
