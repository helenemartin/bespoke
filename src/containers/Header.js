import React from "react";
import { Header } from "../components";
import * as ROUTES from "../constants/routes";
import Navbar from "../components/nav/Navbar";

export function HeaderContainer({ children }) {
  return (
    <>
      <Navbar></Navbar>
      <Header>
        <Header.Frame>
          <Header.Logo
            to={ROUTES.HOME}
            src={process.env.PUBLIC_URL + "/" + "images/dolly.jpg"}
            alt="Bespoke"
          />
          <Header.ButtonLink to={ROUTES.SIGN_IN}>Sign In</Header.ButtonLink>
        </Header.Frame>
      </Header>
    </>
  );
}
