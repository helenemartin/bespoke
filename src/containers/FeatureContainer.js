import React from "react";
import featureData from "../fixtures/feature.json";
import Feature from "../components/feature";
import { FeatureItemContainer } from "./FeatureItemContainer";

export function FeatureContainer() {
  return (
    <Feature.Container>
      {featureData.map((item) => (
        <FeatureItemContainer item={item} key={item.id} />
      ))}
    </Feature.Container>
  );
}
