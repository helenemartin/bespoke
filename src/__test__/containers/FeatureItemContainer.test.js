import { render, screen, fireEvent } from "@testing-library/react";
import { FeatureItemContainer } from "../../containers/FeatureItemContainer";

const mockSpeak = jest.fn();

jest.mock("react-speech-kit", () => ({
  useSpeechSynthesis: () => ({
    speak: mockSpeak,
  }),
}));

describe("FeatureItemContainer Test", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test("should render title", () => {
    // given
    const expectedItem = {
      id: 1,
      title: "One",
      subTitle: "Click the button",
      image: "/images/fleischmann-gusti.jpg",
      alt: "bla",
      direction: "row",
      spokenText: "Hello",
    };

    // when
    const { container } = render(<FeatureItemContainer item={expectedItem} />);

    // then
    expect(container).toMatchSnapshot();
    expect(screen.queryByTestId("title")).toBeInTheDocument();
    expect(screen.getByTestId("title").textContent).toBe(expectedItem.title);
  });

  test("should invoke speak when clicking the item", () => {
    // given
    const expectedItem = {
      spokenText: "Hello world",
    };
    render(<FeatureItemContainer item={expectedItem} />);

    // when
    fireEvent.click(screen.getByTestId("clickable-feature"));

    // then
    expect(mockSpeak).toBeCalledTimes(1);
    expect(mockSpeak).toBeCalledWith({ text: expectedItem.spokenText });
  });
});
