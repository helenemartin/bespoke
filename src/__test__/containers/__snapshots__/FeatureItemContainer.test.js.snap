// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureItemContainer Test should render title 1`] = `
<div>
  <button
    data-testid="clickable-feature"
  >
    <div
      class="sc-dlfnbm dziBQp"
    >
      <div
        class="sc-bdfBwQ gpHtOy"
        direction="row"
      >
        <div
          class="sc-gsTCUz buEYpP"
        >
          <h1
            class="sc-eCssSg hJediz"
            data-testid="title"
          >
            One
          </h1>
          <h2
            class="sc-jSgupP epbPTU"
          >
            Click the button
          </h2>
        </div>
        <div
          class="sc-gsTCUz buEYpP"
        >
          <img
            alt="bla"
            class="sc-gKsewC ngpnn"
            src="/images/fleischmann-gusti.jpg"
          />
        </div>
      </div>
    </div>
  </button>
</div>
`;
